diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/FunctionLength.kt

```diff
@@ -0,0 +1,52 @@
+package org.cqfn.diktat.ruleset.rules
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType.BLOCK_COMMENT
+import com.pinterest.ktlint.core.ast.ElementType.EOL_COMMENT
+import com.pinterest.ktlint.core.ast.ElementType.FUN
+import com.pinterest.ktlint.core.ast.ElementType.KDOC
+import org.cqfn.diktat.common.config.rules.RuleConfiguration
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.common.config.rules.getRuleConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.TOO_LONG_FUNCTION
+import org.cqfn.diktat.ruleset.utils.findAllNodesWithSpecificTypes
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+/**
+ * Rule 5.1.1 check function length
+ */
+class FunctionLength(private val configRules: List<RulesConfig>) : Rule("function-length") {
+
+    companion object {
+        val FUNCTION_ALLOW_COMMENT = listOf(EOL_COMMENT, KDOC, BLOCK_COMMENT)
+    }
+
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        val configuration = FunctionLengthConfiguration(
+                configRules.getRuleConfig(TOO_LONG_FUNCTION)?.configuration ?: mapOf()
+        )
+
+        if (node.elementType == FUN)
+            checkFun(node.copyElement(), configuration.maxFunctionLength)
+    }
+
+    private fun checkFun(node: ASTNode, maxFunctionLength: Long) {
+        node.findAllNodesWithSpecificTypes(FUNCTION_ALLOW_COMMENT).forEach { it.treeParent.removeChild(it) }
```

Comment:
But this will permanently damage AST, won't it?

Reply:
No, because I passed a copy of the node